### PR TITLE
Introduce pegasus-all meta-project

### PR DESCRIPTION
**Pull request by evanw555**
Adds a new module, `pegasus-all`, which is simply a meta-project that
programmatically depends on all other consumer-facing modules. This
makes it easier for certain use cases to pull in the entire dependency
tree.

### CODE REVIEW

File: pegasus-all/build.gradle [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
+    This is a meta-project that programmatically depends on all other consumer-facing modules
```
Comment (junchuanwang) [inline pegasus-all/build.gradle:2]: Should we add any documentations to other places as well?

Comment (evanw555) [reply pegasus-all/build.gradle:2]: I'm not sure. We could add documentation to the website, but there isn't currently documentation on which artifacts do what. We could potentially add documentation to the description of each module's POM artifact, but that's a change for another day.

File: pegasus-all/build.gradle [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    so that the entire dependency tree of pegasus may be pulled in easily. Be warned that this
+    module is not intended to be directly consumed, since it'll bloat the consumer's dependencies.
```
Comment (junchuanwang) [inline pegasus-all/build.gradle:4]: Based on the wording here, it's usage is somewhat non-standard? I will recommend adding to somewhere to let consumer's know.

Comment (evanw555) [reply pegasus-all/build.gradle:4]: I'm not too sure how we can surface this to people. I think I'll merge this for now and ponder that afterward.

**Pull request merged**